R/collect_axes.R

```diff
@@ -34,36 +35,47 @@ collect_axis_titles <- function(gt, dir = "x", merge = TRUE) {
       next
     }
 
-    # Identify 'run'-rectangles in the structure
-    runs <- rle_2d(structure, byrow = dir == "y")
-    runs <- runs[!is.na(runs$value), , drop = FALSE]
-
-    # Find first grob in run
-    start_runs <- c("row_start", "col_start")
-    if (name == "xlab-b") start_runs[1] <- "row_end"
-    if (name == "ylab-r") start_runs[2] <- "col_end"
-    start_idx <- layout[as.matrix(runs[, start_runs])]
+    structure[nested] <- 0
 
-    # Find last grob in run
-    end_runs <- c("row_end", "col_end")
-    if (name == "xlab-t") end_runs[1] <- "row_start"
-    if (name == "ylab-l") end_runs[2] <- "col_start"
-    end_idx <- layout[as.matrix(runs[, end_runs])]
+    # Identify 'run'-rectangles in the structure
+    runs <- rle_2d(structure, byrow = dir == "y", ignore.na = TRUE)
+    runs <- runs[!is.na(runs$value) & runs$value != 0, , drop = FALSE]
+
+    # Get all panels in each run and put the keeper first
+    panels <- lapply(seq_len(nrow(runs)), function(i) {
+      rows <- runs$row_start[i]:runs$row_end[i]
+      cols <- runs$col_start[i]:runs$col_end[i]
+      first <- switch(name,
+        "xlab-t" = layout[runs$row_start[i], cols],
+        "xlab-b" = layout[runs$row_end[i]],
+        "ylab-l" = layout[rows, runs$col_start[i]],
+        "ylab-r" = layout[rows, runs$col_end[i]]
+      )
```

Comment:
I like switch better than the `if` blocks

Reply:
good you commented on that so I saw the missing indexing in the second case 😅